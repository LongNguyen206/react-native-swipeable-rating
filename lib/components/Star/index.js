import React from 'react';
import { TouchableOpacity, } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import PropTypes from 'prop-types';


const Star = ({
  hasOnPress,
  filled,
  size,
  half,
  onPress,
  gap,
  n,
  color,
  emptyColor,
  filledIcon,
  halfFilledIcon,
  emptyIcon, }) => (
    <TouchableOpacity
      disabled={!hasOnPress}
      onPress={() => onPress(n)}
    >
      {typeof filledIcon === 'string' ?
        <Icon
          style={{
            color: filled ? color : emptyColor,
            marginRight: gap
          }}
          name={filled ? (half ? halfFilledIcon : filledIcon) : emptyIcon}
          size={size}
        />
        :
        filled ? (half ? halfFilledIcon(size, gap, n) : filledIcon(size, gap, n)) : emptyIcon(size, gap, n)

      }
    </TouchableOpacity>
);

Star.propTypes = {
  hasOnPress: PropTypes.bool,
  filled: PropTypes.bool,
  size: PropTypes.number,
  half: PropTypes.bool,
  onPress: PropTypes.func,
  gap: PropTypes.number,
  n: PropTypes.number.isRequired,
  color: PropTypes.string,
  emptyColor: PropTypes.string,
  filledIcon: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.function,
  ]),
  halfFilledIcon: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.function,
  ]),
  emptyIcon: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.function,
  ]),
};

export default Star;
