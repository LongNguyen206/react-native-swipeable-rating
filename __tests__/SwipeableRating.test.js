import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import SwipeableRating, { styles } from '../lib/swipeable-rating';
import Star from '../lib/components/Star';

const filledIcon = (size, gap, rating ) =>
  <View style={{width: size, marginRight: gap}}>
    <Text>
      {rating}
    </Text>
  </View>;

const emptyIcon = ( size, gap, rating ) =>
  <View style={{ width: size, marginRight: gap }}>
    <Text>
      {rating}
    </Text>
  </View>;

describe('SwipeableRating Snapshot tests', () => {
  it('renders a 5 SwipeableRating using Snapshots', () => {
    expect(renderer.create(
      <SwipeableRating rating={5} />
    )).toMatchSnapshot();
  });

  it('renders a 3.5 star SwipeableRating using Snapshots', () => {
    expect(renderer.create(
      <SwipeableRating allowHalves rating={3.5} />
    )).toMatchSnapshot();
  });
});
describe('SwipeableRating shallow with Enzyme', () => {
  const wrapper = shallow(
    <SwipeableRating
      swipeable={false}
      rating={4}
      onPress={jest.fn()}
    />
  );

  // console.log(wrapper.debug());
  // console.log(wrapper.props());

  it('renders the correct filled/unfilled stars', () => {
    expect(
      wrapper.containsAllMatchingElements([
        <Star n={1} hasOnPress={true} filled={true} half={false} />,
        <Star n={2} hasOnPress={true} filled={true} half={false} />,
        <Star n={3} hasOnPress={true} filled={true} half={false} />,
        <Star n={4} hasOnPress={true} filled={true} half={false} />,
        <Star n={5} hasOnPress={true} filled={false} half={false} />
      ])).toBe(true);
  });

});

describe('SwipeableRating mounted with Enzyme', () => {
  
  const wrapper = mount(
    <SwipeableRating
      swipeable={false}
      rating={4}
      filledIcon={filledIcon}
      emptyIcon={emptyIcon}
    />
  );
    // console.log(wrapper.debug())
  it('renders custom icon components', () => {
  expect(
    wrapper.containsAllMatchingElements(
      [filledIcon(24, 0, 1), emptyIcon(24, 0, 1)]
    )).toBe(true);
  });
});