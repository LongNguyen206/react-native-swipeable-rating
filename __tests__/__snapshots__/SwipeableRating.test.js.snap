// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableRating Snapshot tests renders a 3.5 star SwipeableRating using Snapshots 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`SwipeableRating Snapshot tests renders a 5 SwipeableRating using Snapshots 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "crimson",
            "marginRight": 0,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
